test/http/http.go

```diff
@@ -109,6 +109,18 @@ func CaptureRoundTrip(method, scheme, hostname, path, location string) (*Capture
 	}
 	defer resp.Body.Close()
 
+	// check if the result is a redirect and return a new request
+	// this avoids the issue of URLs without valid DNS names and
+	// also sends the traffic to the ingress controller IP address or FQDN
+	if isRedirect(resp.StatusCode) {
+		redirectURL, err := resp.Location()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return CaptureRoundTrip(method, redirectURL.Scheme, redirectURL.Hostname(), redirectURL.Path, location)
```

Comment:
I would also add a flag to the `CapturedResponse` to identify which requests initial response was a redirect. We could then assert whether a redirect was made in features that are expecting it (http -> https)

Reply:
+1